src/main/java/org/veupathdb/service/eda/ds/plugin/AbstractPlugin.java

```diff
@@ -610,4 +626,53 @@ public String getRBinListAsString(List<String> labels) {
     return rBinList + "))";
   }
 
+  public boolean validateImputeZeroesRequest(Map<String, VariableSpec> varSpecs) {
+    // TODO check a bunch of stuff i no longer remember
+    return true;
+  }
+
+  public String getRStudyVocabsAsString(VocabByRootEntityPostResponse studyVocab) {
+    // TODO make a string to build the vocabs slot
+  }
+
+  public Strng getRStudyVocabsAsString(List<VocabByRootEntityPostResponse> studyVocabs) {
+    // TODO loop through and call the other version once for each entry. sandwich that in a R list.
+  }
```

Comment:
I'm rolling off some other work. If you want I can take a stab at filling in the AbstractPlugin methods. What's the desired string format here though? I'm having trouble conceptualizing it since the vocab is two dimensional

Reply:
this is still wip, and i dont think will be awful to do. i just need to get to it. if youre in need of work, mbio is looking for help still.